airflow/contrib/hooks/gcp_dlp_hook.py

```diff
@@ -0,0 +1,1362 @@
+        :rtype: google.cloud.dlp_v2.types.DeidentifyTemplate
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException("Please provide the deidentifyTemplate name.")
+
+        return client.get_deidentify_template(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def get_dlp_job(self, name, retry=None, timeout=None, metadata=None):
+        """
+        Gets the latest state of a long-running DlpJob.
+
+        :param name: The name of the DlpJob resource.
+        :type name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        :rtype: google.cloud.dlp_v2.types.DlpJob
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException("Please provide the name of the DlpJob resource.")
+
+        return client.get_dlp_job(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def get_inspect_template(self, name, retry=None, timeout=None, metadata=None):
+        """
+        Gets an InspectTemplate.
+
+        :param name: Resource name of the organization and inspectTemplate to be read.
+        :type name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        :rtype: google.cloud.dlp_v2.types.InspectTemplate
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException("Please provide the inspectTemplate name.")
+
+        return client.get_inspect_template(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def get_job_trigger(self, name, retry=None, timeout=None, metadata=None):
+        """
+        Gets a job trigger.
+
+        :param name: Resource name of the project and the triggeredJob.
```

Comment:
It is worth adding the possibility of defining the project ID using the connection configuration.